clients/drcachesim/tools/view.cpp

```diff
@@ -314,142 +322,35 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
-                std::cerr << "<marker: function==syscall #"
-                          << (memref.marker.marker_value -
-                              static_cast<uintptr_t>(
-                                  func_trace_t::TRACE_FUNC_ID_SYSCALL_BASE))
-                          << ">\n";
-            } else {
-                std::cerr << "<marker: function #" << memref.marker.marker_value << ">\n";
-            }
-            break;
-        case TRACE_MARKER_TYPE_FUNC_RETADDR:
-            std::cerr << "<marker: function return address 0x" << std::hex
-                      << memref.marker.marker_value << std::dec << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_FUNC_ARG:
-            std::cerr << "<marker: function argument 0x" << std::hex
-                      << memref.marker.marker_value << std::dec << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_FUNC_RETVAL:
-            std::cerr << "<marker: function return value 0x" << std::hex
-                      << memref.marker.marker_value << std::dec << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_SYSCALL_FAILED:
-            std::cerr << "<marker: system call failed: " << memref.marker.marker_value
-                      << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_RECORD_ORDINAL:
-            std::cerr << "<marker: record ordinal 0x" << std::hex
-                      << memref.marker.marker_value << std::dec << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_SYSCALL:
-            std::cerr << "<marker: system call " << memref.marker.marker_value << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL:
-            std::cerr << "<marker: maybe-blocking system call>\n";
-            break;
-        case TRACE_MARKER_TYPE_DIRECT_THREAD_SWITCH:
-            std::cerr << "<marker: direct switch to thread " << memref.marker.marker_value
-                      << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_WINDOW_ID:
-            // Handled above.
-            break;
-        case TRACE_MARKER_TYPE_SYSCALL_TRACE_START:
-            std::cerr << "<marker: trace start for system call number "
-                      << memref.marker.marker_value << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_SYSCALL_TRACE_END:
-            std::cerr << "<marker: trace end for system call number "
-                      << memref.marker.marker_value << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_CONTEXT_SWITCH_START:
-            std::cerr << "<marker: trace start for context switch type "
-                      << memref.marker.marker_value << ">\n";
-            break;
-        case TRACE_MARKER_TYPE_CONTEXT_SWITCH_END:
-            std::cerr << "<marker: trace end for context switch type "
+        case TRACE_MARKER_TYPE_SPLIT_VALUE:
+        case TRACE_MARKER_TYPE_SYSCALL_IDX:
+            std::cerr << "<marker: type " << memref.marker.marker_type << "; value "
```

Comment:
Seems like these 2 cases can be omitted as this is the default behavior in the marker printing helper called below.

Reply:
We actually catch those two cases in `trace_marker_type_value_as_string()` and print a more informative output that looks different than the one here: <marker: type NUM; value NUM>.
So, to preserve the previous view.cpp behavior, we catch these 2 cases early and print them the same way view.cpp used to print them.

We can remove these two cases if:
1) we decide we don't care about printing a more informative output (hence we remove them from here and `trace_marker_type_value_as_string()`, letting the default in `trace_marker_type_value_as_string()` take care of it. (<--- not a fan of this, I think we should print the most "informative" version of marker type and value)
2) we decide to update view.cpp's printing for those two cases and align it to `trace_marker_type_value_as_string()`. (<--- might break doc and examples?)

Reply:
SPLIT_VALUE will never be seen by view.cpp as it is never presented as a memref_t, so there is no reason to have an explicit case entry for it.

Anything not explicitly in a case statement in view.cpp can be considered to not matter: if someone wanted to see that marker we would add a case statement.  The `default` there is a fallback and we do not need to preserve it: we don't want to preserve it: we want to print the string name, not the marker enum number.

Reply:
Got it.
Then, since those two cases were both catched in what used to be view.cpp's default case, we can remove them and let `trace_marker_type_value_as_string()` print them (even though view.cpp will never see SPLIT_VALUE).
Going with 2).